catalog/app/utils/gtm.tsx

```diff
@@ -0,0 +1,40 @@
+import * as React from 'react'
+
+import cfg from 'constants/config'
+
+function loadScript(src: string) {
+  const s = window.document.createElement('script')
+  s.type = 'text/javascript'
+  s.async = true
+  s.src = src
+
+  const x = window.document.getElementsByTagName('script')[0]
+  x.parentNode?.insertBefore(s, x)
+}
+
+function addScript(content: string) {
+  const s = window.document.createElement('script')
+  s.type = 'text/javascript'
+  s.innerText = content
+  const x = window.document.getElementsByTagName('script')[0]
+  x.parentNode?.insertBefore(s, x)
+}
+
+interface GTMLoaderProps {
+  children: React.ReactNode
+}
+
+export default function GTMLoader({ children }: GTMLoaderProps) {
+  const gtmId = cfg.gtmId
```

Comment:
maybe it would make sense to get this as a prop, and pass it in `app.tsx`, but... not sure what that would buy us)

Reply:
I don't know how to explain it clearly, but I feel it would be better as well